src/server/snap.rs

```diff
@@ -403,14 +409,19 @@ fn recv_snap<R: RaftExtension + 'static>(
                 return Err(e);
             }
         }
-        // Notify the snapshot manager that a snapshot has been received,
-        // freeing up the associated resource in the concurrency limiter.
-        snap_mgr.recv_snap_complete(context.raft_msg.region_id);
         context.finish(raft_router)
     };
     async move {
+        defer!(cleanup_after_recv(
```

Comment:
Why not clean up before responding to the sink?

Reply:
Discussed with @Connor1996 offline. I believe cleaning up _before_ responding to the sink is still an option but it probably doesn't make a significant difference since responding to the sink should be quick. I think I’ll keep it as it is to maintain consistency with the current behavior, where we decrement `recving_count` _after_ responding to the sink.